arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowVectorAccessors.java

```diff
@@ -31,7 +33,7 @@ final class ArrowVectorAccessors {
   static {
     factory =
         new GenericArrowVectorAccessorFactory<>(
-            throwingSupplier("Decimal type is not supported"),
+            JavaDecimalFactory::new,
             JavaStringFactory::new,
             throwingSupplier("Struct type is not supported"),
             throwingSupplier("List type is not supported"));
```

Comment:
Curious is  there an open issue or existing PR on support for struct/list types? I wasn't able to find any

Reply:
there is https://github.com/apache/iceberg/issues/2485. Additionaly there is also all the issues with the `arrow` label: https://github.com/apache/iceberg/issues?q=is%3Aopen+is%3Aissue+label%3Aarrow

Reply:
Thanks missed this!